elastic/kibana
# [Serverless] Add internal uiSettings routes and expose public routes in self-managed only

Partially addresses https://github.com/elastic/kibana/issues/159590

## Summary

This PR adds an an internal uiSettings API that is a duplicate of the public API and is intended for use by the browser-side uiSettings client.

The PR also adds a config settings that is configured in serverless context only and exposes the public uiSettings routes based on the value of this setting (it defaults to false since we don't want to expose the public routes in serverless).

**How to test:**

I. Verify that in serverless the internal routes are exposed but the public ones aren't:
1. Start Es with `yarn es snapshot` and Kibana with `yarn serverless-{mode}` where `{mode}` can be `es`, `oblt`, or `security` (the public routes should be disabled for all projects).
2. Verify that the public endpoints are not accessible. For example, `curl --user elastic:changeme 'http://localhost:5601/zhb/api/kibana/settings' -X 'GET'` should return `{"statusCode":404,"error":"Not Found","message":"Not Found"}`.
3. Verify that the internal endpoints are accessible. For example, `curl --user elastic:changeme 'http://localhost:5601/zhb/internal/kibana/settings' -X 'GET'` should return `{"settings":{"buildNum":{"userValue":9007199254740991},"isDefaultIndexMigrated":{"userValue":true},"defaultRoute":{"isOverridden":true,"userValue":"/app/elasticsearch"}}}`


II. Verify that the both public and internal routes are exposed in self-managed:
1. Start Es with `yarn es snapshot` and Kibana with `yarn start`
2. Verify that the public endpoints are accessible. For example, `curl --user elastic:changeme 'http://localhost:5601/zhb/api/kibana/settings' -X 'GET'` should return `{"settings":{"buildNum":{"userValue":9007199254740991},"isDefaultIndexMigrated":{"userValue":true}}}`
3. Verify that the internal endpoints are accessible. For example, `curl --user elastic:changeme 'http://localhost:5601/zhb/internal/kibana/settings' -X 'GET'` should return `{"settings":{"buildNum":{"userValue":9007199254740991},"isDefaultIndexMigrated":{"userValue":true}}}`

III. Verify that the plugins/services that consume the internal uiSettings endpoints work as expected in both self-managed and serverless environment.

### For maintainers
